res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -0,0 +1,2254 @@
+        engine.softTakeover(group, "pitch_adjust", true);
+        engine.softTakeover(group, "volume", true);
+        engine.softTakeover(group, "pregain", true);
+        engine.softTakeover("[QuickEffectRack1_" + group + "]", "super1", true);
+    }
+    for (var unit = 1; unit <= 4; unit++) {
+        group = "[EffectRack1_EffectUnit" + unit + "]";
+        var key = "mix";
+        engine.softTakeover(group, key, true);
+        for (var effect = 1; effect <= 4; effect++) {
+            group = "[EffectRack1_EffectUnit" + unit + "_Effect" + effect + "]";
+            key = "meta";
+            engine.softTakeover(group, key, true);
+            for (var param = 1; param <= 4; param++) {
+                key = "parameter" + param;
+                engine.softTakeover(group, key, true);
+            }
+        }
+    }
+
+    engine.softTakeover("[Microphone]", "volume", true);
+    engine.softTakeover("[Microphone]", "pregain", true);
+
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter1", true);
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel2]_Effect1]", "parameter1", true);
+    engine.softTakeover("[EqualizerRack1_[Channel2]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel2]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel3]_Effect1]", "parameter1", true);
+    engine.softTakeover("[EqualizerRack1_[Channel3]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel3]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel4]_Effect1]", "parameter1", true);
+    engine.softTakeover("[EqualizerRack1_[Channel4]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel4]_Effect1]", "parameter3", true);
+
+    // engine.softTakeover("[Master]", "crossfader", true);
+    engine.softTakeover("[Master]", "gain", true);
+    // engine.softTakeover("[Master]", "headMix", true);
+    // engine.softTakeover("[Master]", "headGain", true);
+
+    for (var i = 1; i <= 16; ++i) {
+        engine.softTakeover("[Sampler" + i + "]", "pregain", true);
+    }
+
+    for (ch in this.Channels) {
+        var chanob = this.Channels[ch];
+        engine.makeConnection(ch, "playposition",
+            TraktorS3.bind(TraktorS3.Channel.prototype.playpositionChanged, chanob));
+        engine.connectControl(ch, "track_loaded",
+            TraktorS3.bind(TraktorS3.Channel.prototype.trackLoadedHandler, chanob));
+        engine.connectControl(ch, "end_of_track",
+            TraktorS3.bind(TraktorS3.Channel.prototype.endOfTrackHandler, chanob));
+    }
+
+    // Dirty hack to set initial values in the packet parser
+    var data = [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+    TraktorS3.incomingData(data);
+};
```

Comment:
When #3317 is merged, you could use controller.getInputReport(0x01); to initialize data with the real knob positions.